### PR TITLE
Remove use of `softmax` in example of `torch.nn.functional.cross_entropy`

### PR DESCRIPTION
**Pull request by Theo-Cheynel**
The cross-entropy (as implemented in `torch.nn.functional.cross_entropy`) takes as input un-normalized logits, and this is specified in the docstring.

However, the examples at the bottom of the docstring still apply a softmax before calling the function, which effectively leads to two softmax being applied in a row, thus yielding terrible results.

I suggest we update the last example to prevent anyone from thinking that they have to apply softmax before cross-entropy.

### CODE REVIEW

**Review by mikaylagawarecki — COMMENTED** [review]
The `softmax` is applied in `target` not `input`. Per the docs for CrossEntropyLoss, `target` can either be indices or class probabilities, 

`torch.randn(3, 5).softmax(dim=1)` gives probabilities as stated in the comment `# Example of target with class probabilities`, so I don't think we want to make this change

**Comment by Theo-Cheynel** [comment]
Well, except that the cross-entropy loss is not supposed to take probabilities, it is supposed to take logits, since it applies a softmax. This is well reflected in the docstrings of the CE function and class, for instance : "The input is expected to contain the unnormalized logits for each class (which do not need to be positive or sum to 1, in general).")

If you look at the C++ code (https://github.com/pytorch/pytorch/blob/2981534f54d49fa3a9755c9b0855e7929c2527f0/aten/src/ATen/native/LossNLL.cpp#L568), you'll see that it already uses logsoftmax.

Here is the problem though : say I have three logits, and they have values [-10, -10, 3]. If my model has a softmax layer, this output becomes very close to [0, 0, 1].
However, since the CELoss uses another softmax, the values become [0.21, 0.21, 0.58]. If my expected distribution was for instance [0, 0, 1] (or simply giving the index 2) we see that the consecutive softmaxes cause an issue.

So I stand by what I said, the example in this docstring goes against what was said previously in said docstrings, as well as other docstrings.
